### PR TITLE
LPS-196687 Add KeyboardDragAndDropContext

### PR DESCRIPTION
**Pull request by p2kmgcl**
`LPS-177686 *` commits are from https://github.com/liferay-echo/liferay-portal/pull/13744 PR, which is already being forwarded.

This PR needs `LPS-196420` feature flag to be tested.

It adds an initial version of keyboard DND, but still needs some things to be polished. I do want to send some work to master so I don't end up with a mega-PR.

What is included in this PR:
- Components have been moved and split into separated files in a new directory.
- Mouse-related dnd has been extracted to a context file (although technically it does not have it's own context :sweat_smile:).
- A new KeyboardDNDContext has been created. It's current behavior is:
  - Navigate to a drag handler using tab key.
    > This will be updated in a future PR, users should tab just to the "list" component, and navigation should happen with up/down arrow keys.
  - Press space to start dragging the current item.
  - Move it to the desired position using up/down arrow keys.
  - Press Escape to cancel the drag, or Enter to drop it.

> I haven't added any screen reader feedback, will do in a different PR.

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Review by sandrodw3 — COMMENTED** [review]
Hey @p2kmgcl this is awesome as usual 🤍 

Just left a tiny comment regarding the code, and I have also some suggestions/questions about the functionality:
- It's possible to finish the keyboard movement with Enter, but it's not possible to start it with Enter, we can only start it with Space, is that intended? For other keyboard DnD I think we are supporting both
- As we are using click event on the drag button, I found a weird behavior that is caused when clicking it with mouse. It starts the keyboard movement, what I think can be a bit confusing for the user. In other cases we are only making the handler visible when it's focused, I don't know if it's a valid option for this case 🤔

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments_and_widgets/components/reorder_sets_modal/Item.tsx [hunk 1]
```diff
@@ -66,16 +69,38 @@ export function Item({index, item, numberOfItems, onDropItem}: ItemProps) {
...
+										<ClayIcon
+											className="text-secondary"
+											symbol="drag"
+										/>
+
+										<span
+											className="sr-only"
+											id={dragButtonDescriptionId}
+										>
+											{Liferay.Language.get('reorder')}
```
[the remark this answers is not in the document]
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments_and_widgets/components/reorder_sets_modal/Item.tsx:110]: It is, that's why the button (above) has the icon **and** the item description as labelledby :smile:

**Comment by p2kmgcl** [comment]
> It's possible to finish the keyboard movement with Enter, but it's not possible to start it with Enter, we can only start it with Space, is that intended? For other keyboard DnD I think we are supporting both

I'll fix that!

**Comment by p2kmgcl** [comment]
> As we are using click event on the drag button, I found a weird behavior that is caused when clicking it with mouse. It starts the keyboard movement, what I think can be a bit confusing for the user. In other cases we are only making the handler visible when it's focused, I don't know if it's a valid option for this case 🤔

I think I am going to remove the click handler and only listen keyboard events :thinking:. What do you think about that?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments_and_widgets/components/reorder_sets_modal/Item.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sandrodw3) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/fragments_and_widgets/components/reorder_sets_modal/Item.tsx:110]: Ah! So sorry 😅 I just saw it, too fast the first time

**Comment by p2kmgcl** [comment]
@sandrodw3 done! just the last three commits

**Comment by sandrodw3** [comment]
@p2kmgcl everything is working perfectly now 🥳 I just found a weird behavior with the scroll, don't know if it's related to the new commits. When moving from last position to top, it scrolls up a bit and then scrolls down again, I'm attaching a video. Feel free to ignore this if you don't find it annoying 😛

https://github.com/liferay-echo/liferay-portal/assets/2444936/70dc935c-f30c-4d99-96a6-56fedbf661ef

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by liulukiwi** [comment]
ci:test:bundle

**Comment by liulukiwi** [comment]
ci:stop

**Comment by liulukiwi** [comment]
Hi @p2kmgcl , there's an issue found in this pull, can you take a look? thanks! here are the steps:
1. Use the following command to enable BasicWebContentSingleFormVariationInfoCollectionProvider via gogo shell
scr:enable com.liferay.journal.web.internal.info.collection.provider.BasicWebContentSingleFormVariationInfoCollectionProvider
2. Add a new site
3. Add a basic web content 
4. Add a Collection Display to a content page
5. Select the Basic Web Content collection provider
6. Open the Filter Collection modal via dropdown menu of Collection field

Actual result:
No items display.
fail

Expected result:
Tags and Title should display.
expected

**Comment by p2kmgcl** [comment]
https://github.com/liferay-echo/liferay-portal/pull/13835

**Pull request closed without merging**